### PR TITLE
Lerna setup

### PR DESCRIPTION
**Pull request by relekang**

### CODE REVIEW

**Comment by havardh** [comment]
The failure in the tests are due to the `yarn` command now generating `dist` folders. 
I think we should change the test to look at a folder that will not depend on the build process. So lets make it look at `workflow-resolver-relative/src` instead. :)

**Comment by havardh** [comment]
I'm not seeing the expected results when running `lerna changed`/`lerna version` on this branch. 

Lets take the `workflow` package as an examples. It was release as `1.0.0` in `680623a` tagged with `workflow-v1.0.0`. `git log workflow-v1.0.0..HEAD -- packages/workflow` is empty. So I'm expecting no changed when using `lerna`.

However, `lerna changed` mentions `workflow` and `lerna version` want's to bump it to `1.1.0`. `lerna diff workflow` comes up empty. 

So I looked into the dependencies of `workflow`, which are `workflow-core`, `workflow-react` and `workflow-cmd`. Of these, only `workflow-core` has any changes, which is a `chore` at `14ae8c63`. 

Now at the top of the `lerna version`/`lerna changed` output is the following line: 
```
lerna info Looking for changed packages since workflow-app-tmux-v0.1.1
```
And given that fact, the rest of the output could make sense. 

Any idea why it choose this tag as the "previous" release?

**Comment by relekang** [comment]
>So I looked into the dependencies of workflow, which are workflow-core, workflow-react and workflow-cmd. Of these, only workflow-core has any changes, which is a chore at 14ae8c63.

I think this makes sense, since lerna will release anything depending on something getting released and it seems there is no option to ignore chores.

> Any idea why it choose this tag as the "previous" release?

Based on the output from `git for-each-ref refs/tags` it looks like it is the last annotated tags. Did the previous process move away from annotated tags to lightweight tags(seen as commit in the output of the command I mentioned).


This gets us to something I have been thinking about. Many of the apps packages have dependencies of the example set as dependencies and not dev-dependencies. Can we change that?

**Comment by havardh** [comment]
Ah, great :) I'll try again with annotated tags and that flag when I have time :)

We have to clean up the dependencies ref #120. The question is if we should hoist all to the root repository or maintain them as devDependencies. My argument for the latter is that it would make it easier to find examples for someone wanting to develop a package outside of this repository, as all packages would be self contained. I do however have a plan for making `create-workflow-*` which should include the dependencies if it run outside of this repository. 

What do you think about only maintaing them at the root? We do not at this point have any binaries below the root that would require them to be local.

**Comment by relekang** [comment]
I think it is a good idea to try to have them as dev dependencies to be explicit about it.

**Comment by havardh** [comment]
TL;DR; just my notes to figure out why all these packages bumped a minor version. Skip ahead to the conclusion/question at the bottom. 

So I've updated the tags to be annotated and now `lerna version` is picking the expected last tag `create-workflow-app-v1.0.1`. The output seems to make sense for what has updates since this tag: 

```
 - workflow-app-atom: 0.2.1 => 0.3.0
 - workflow-app-chrome: 0.2.0 => 0.3.0
 - workflow-app-code: 0.2.0 => 0.3.0
 - workflow-app-emacs: 0.2.0 => 0.3.0
 - workflow-app-less: 0.2.0 => 0.3.0
 - workflow-app-notepad: 0.2.0 => 0.3.0
 - workflow-app-osxterminal: 0.2.0 => 0.3.0
 - workflow-app-powershell: 0.2.0 => 0.3.0
 - workflow-app-safari: 0.2.1 => 0.3.0
 - workflow-app-ssh: 0.2.0 => 0.3.0
 - workflow-app-terminal: 0.2.1 => 0.3.0
 - workflow-app-tmux: 0.3.0 => 0.4.0
 - workflow-app-vim: 0.2.1 => 0.3.0
 - workflow-app-xterm: 0.2.0 => 0.3.0
 - workflow-apps-defaults: 0.3.2 => 0.4.0
 - workflow-layout-tiled: 1.0.1 => 1.1.0
 - workflow-layout-yoga: 0.2.0 => 0.3.0
 - workflow-resolver-relative: 1.0.0 => 1.1.0
 - workflow-wm-auto: 0.3.2 => 0.4.0
 - workflow-wm-wmctrl: 0.2.0 => 0.3.0
```
Based on the following change log:

```
chore(flow): Add ignore of tests files from node_modules
f29aa37 test: Use fixtures in workflow-resolver-relative
40ed63a chore: Upgrade some workflow packages in workflow packages
eaa3c78 chore: Add lerna configuration
026345d test: update snapshot for windows
e395b54 fix: make powershell stay open when cmd is not given
e1a4664 feat: add support for linux-i3 for vs code
5f1971b test: update snapshot for osx
b2a55f5 test: make iterm cursor independent of focus
```

I was expecting:

```
 - workflow-app-code: 0.2.0 => 0.3.0
 - workflow-app-powershell: 0.2.0 => 0.2.1
```
And I want those + their direct dependencies to be updated. 

Now it seems like either the `chore` or the `test` commits are picked up as a minor bump. And the rest of the packages changes because they have a `dependency` or `devDependency` one another package which changes. 

- - -

So the question remains if we should remove the devDepenencies from the packages, so that transient bumps will only happend for actual dependencies. 

And if we are able to configure the version bumps to ignore tests and chores. `lerna` uses the `conventinal-recommended-bump` package, but I'm unsure if we are able to pass configuration to it.

**Comment by havardh** [comment]
We can use this to fix the `test` commits https://github.com/lerna/lerna/issues/1554

**Comment by havardh** [comment]
And a discussion on `chore` https://github.com/lerna/lerna/issues/1425 -> https://github.com/conventional-changelog/standard-version/issues/163.

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "conventionalCommits": true,
+  "useWorkspaces": true,
+  "packages": [
+    "packages/*"
+  ],
+  "ignoreChanges": [
+    "**/__fixtures__/**",
+    "**/__tests__/**",
+    "**/test/**",
+    "**/*.md"
```
Comment (relekang) [inline lerna.json]: I am a bit thorn about this. We might want changes to readmes to be released to be reflected on npm. @havardh What do you think?

Comment (havardh) [reply lerna.json]: Yes we want that. I'm not sure about the `__tests__` or the `__fixtures__` pattern. I did just merge the `__fixtures__` thing so we'll have to live with that for now.

Comment (relekang) [reply lerna.json]: The __fixtures__ looks nice next to __snapshots__ 😛
